lib/apiservers/engine/backends/filter/container.go

```diff
@@ -87,31 +84,59 @@ func IncludeContainer(listContext *ContainerListContext, container *models.Conta
 	}
 
 	// filter on network name
-	for n := range container.Endpoints {
-		name := container.Endpoints[n].Scope
-		if listContext.Filters.Include("network") {
-			err := listContext.Filters.WalkValues("network", func(value string) error {
-				if name == value {
-					return nil
-				}
-				return fmt.Errorf("not part of container network")
-			})
-			if err != nil {
-				return ExcludeAction
+	if listContext.Filters.Include("network") {
+		netFilterValues := listContext.Filters.Get("network")
+
+		// Gather the container's networks in a map
+		numNetworks := len(container.Endpoints)
```

Comment:
networks and volumes act as an "or" filter for docker -- so if the container has _any_ of the filter values then it should be **included** in the return.  The current code will exclude containers that should be included in both volume and network filtering.

I suggest you loop over the filters -- as soon as a filter is value found break the loop and that container should be **included** in the results -- if no filter values are found then exclude it.  I would do this without the creation of the volume / network maps -- in both cases just loop over the filters and then loop over the networks / volumes.  I'm expecting there will typically be few filters, so iterating of the collections is fine.